treetoreads.py

```diff
@@ -687,104 +869,72 @@ def run_art(self, coverage=None):
-                self._exit_handler()
+            for seqnam in self.seqnames:
+                real_len = random.sample(self.argdict['read_length'], 1)[0] 
+                self.read_spec[seqnam] = {'cov' : self.argdict['coverage'], 'rl' : real_len}
+        if self.argdict['fragment_size'] is None:
+            frag_low = int(self.argdict['frag_low'])
+            frag_hi = int(self.argdict['frag_high'])
+            frag_step = int(self.argdict['frag_step'])
+            for seqnam in self.seqnames:
+                real_frag = random.sample(range(frag_low, frag_hi, frag_step), 1)[0]
+                self.read_spec[seqnam].update({'frag' : real_frag})
         else:
             for seqnam in self.seqnames:
-                cov[seqnam] = int(coverarg)
-        if not os.path.isdir("{}/fastq".format(self.outd)):
-            os.mkdir("{}/fastq".format(self.outd))
-        sys.stdout.write("coverage is {}\n".format(self.config['coverage']))
-        if 'read_length' in self.config:
-            read_length = self.config['read_length']
+                self.read_spec[seqnam].update({'frag' : self.argdict['fragment_size']})
+        if self.argdict['stdev_frag_size'] is None:
+            stdev_low = int(self.argdict['stdev_frag_low'])
+            stdev_hi = int(self.argdict['stdev_frag_high'])
+            stdev_step = int(self.argdict['stdev_frag_step'])
+            for seqnam in self.seqnames:
+                real_stdev = random.sample(range(stdev_low, stdev_hi, stdev_step), 1)[0]
+                self.read_spec[seqnam].update({"stdev" : real_stdev})
         else:
-            read_length = 150
-        sys.stdout.write("read length is {}\n".format(read_length))
-        if 'fragment_size' in self.config:
-            fragment_size = self.config['fragment_size']
+            for seqnam in self.seqnames:
+                self.read_spec[seqnam].update({'stdev' : self.argdict['stdev_frag_size']})
+        if self.argdict['qs2'] is not None:
+            qs2_low = int(self.argdict['qs2_low'])
+            qs2_hi = int(self.argdict['qs2_high'])
+            qs2_step = int(self.argdict['qs2_step'])
+            for seqnam in self.seqnames:
+                real_qs2 = random.sample(range(qs2_low, qs2_hi, qs2_step), 1)[0]
+                self.read_spec[seqnam].update({'qs2' : real_qs2,})
         else:
-            fragment_size = 350
-        sys.stdout.write("fragment size is {}\n".format(fragment_size))
-        if 'stdev_frag_size' in self.config:
-            stdev_frag_size = self.config['stdev_frag_size']
+            for seqnam in self.seqnames:
+                self.read_spec[seqnam].update({'qs2' : 0})
+        for seqnam in self.seqnames:
+            real_platform = random.sample(self.argdict['readProfile'], 1)[0]
+            self.read_spec[seqnam].update({'ver' : real_platform})
+        if not os.path.isdir("{}/fastq".format(self.outd)):
+            os.mkdir("{}/fastq".format(self.outd))
+        if self.threads is None:
```

Comment:
Changing this line to: will make it python2 compatible.
if self.threads is None or sys.version_info[0] < 3: